HashRegistrar.sol

```diff
@@ -0,0 +1,282 @@
+// hash registrar using deposits
+
+contract Deed {
+    // The Deed is a contract intended simply to hold ether
+    // It can be controlled only by the registrar and can only send ether back to the owner
+    Registrar public registrar;
+    bytes32 public sealedBid;
+    bytes32 public hash;
+    address constant burn = 0xdead;
+    uint public creationDate;
+    address public owner;
+  
+    modifier noEther {
+        if (msg.value > 0) throw;
+        _
+    } 
+    
+    modifier onlyRegistrar {
+        if (msg.sender != address(registrar)) throw;
+        _
+    }
+    
+    function Deed(bytes32 _sealedBid, address _registrar) {
+        registrar = Registrar(_registrar);
+        sealedBid = _sealedBid;
+        creationDate = now;
+    }
+    
+    function unsealBid(bytes32 _hash, address _owner, uint _value, bytes32 _salt) noEther onlyRegistrar {
+        if (sha3(_hash, _owner, _value, _salt) != sealedBid) throw;
+        hash = _hash;
+        owner = _owner;
+        if (!owner.send(this.balance - _value)) throw;
+    }
+    
+    function closeDeed(uint refundRatio) noEther onlyRegistrar {
+        owner.send((refundRatio * this.balance)/1000);
+        selfdestruct(burn);
+    }
+    
+    function setBalance(uint newValue) onlyRegistrar {
+        // Check if it has enough balance to set the value
+        if (this.balance < newValue) throw;
+        // Send the difference to the owner
+        if (!owner.send(this.balance - newValue)) throw;
+    }
+    
+    function transferOwnership(address newOwner) {
+        if (msg.sender != owner ) throw;
+        owner = newOwner;
+    }
+}
+
+contract Registrar {
+    mapping (bytes32 => entry) public hashes;
```

Comment:
Or rename entry as hash? Not sure which would make the code clearer. Maybe the word "hash" is already being overused but at least it's more specific than entry..